src/main/java/ui/Ui.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * Gets user inputs from the terminal.
+     */
+    public void getInput() {
+        while (!parser.isBye()) {
+            if (!scanner.hasNext()) {
+                continue;
+            }
+            try {
+                parser.parse(scanner.nextLine());
+            } catch (AliceException e) {
+                System.out.println(e);
+                showDivider();
+            }
+        }
+    }
+
+    /**
+     * Prints message to terminal when task is marked or unmarked.
+     *
+     * @param task the task which was handled.
+     * @param command "mark" or "unmark".
+     */
+    public void printHandleTaskMessage(Task task, String command) {
+        if (command.equals("mark")) {
+            showDivider();
+            System.out.println("Noted! I've marked this task as done: ");
+            System.out.println(task);
+            showDivider();
+        } else {
+            showDivider();
+            System.out.println("Ok, I've marked this task as not done yet: ");
+            System.out.println(task);
+            showDivider();
+        }
+    }
+
+    /**
+     * Prints message to the terminal when task is deleted or added.
+     *
+     * @param task the task that was handled.
+     * @param command "delete" or "todo" or "deadline" or "event".
+     * @param size number of tasks in the list.
+     */
+    public void printHandleTaskMessage(Task task, String command, int size) {
+        if (command.equals("delete")) {
+            showDivider();
+            System.out.println("Noted. I've removed this task: ");
+            System.out.println(task);
+            System.out.printf("Now you have %d tasks in the list%n", size);
+            showDivider();
+        } else {
+            showDivider();
+            System.out.println("Got it. I've added this task: ");
+            System.out.println(task);
+            System.out.printf("Now you have %d tasks in the list%n", size);
+            showDivider();
+        }
+    }
+
+    /**
+     * Prints all the tasks in the list.
+     *
+     * @param tasks the string representation of all the tasks.
+     * @param size number of tasks in the list.
+     */
+    public void printListedTasks(String tasks, int size) {
+        showDivider();
+        System.out.println("Here are the tasks in your list: ");
+        System.out.println(tasks);
+        System.out.printf("Now you have %d tasks in the list%n", size);
+        showDivider();
+    }
+
+    /**
+     * Prints filtered tasks to the terminal.
+     *
+     * @param tasks String of tasks to be displayed.
+     */
+    public void printFilteredTasks(String tasks) {
+        showDivider();
+        System.out.println("Here are the matching tasks in your list: ");
+        System.out.println(tasks);
+        showDivider();
+    }
+
+    /**
+     * Prints the divider to the terminal.
+     */
+    public void showDivider() {
+        System.out.println(Ui.DIVIDER);
+    }
+}
```

Comment:
The Ui class is well-structured and easy to read, with each method having a clear and focused responsibility. The separation of concerns is well-executed, making the code intuitive to understand and maintain. I particularly appreciate how methods like printHandleTaskMessage and showDivider are used to encapsulate specific actions, ensuring that the code remains organised and avoiding repetitive code. This design enhances the overall readability and makes it easy to extend or modify the class in the future. Great job!